openfoodfacts-implementation/data-generation.ipynb

```diff
@@ -0,0 +1,299 @@
+    "        print(recipe_name, recipe_info['nutrition'])\n",
+    "        recipe_info['nutrition']['sugars_100g'] = recipe_info['sugars']\n",
+    "        recipe_info['nutrition']['fiber_100g'] = recipe_info['fiber']\n",
+    "        recipe_info['nutrition']['fat_100g'] = recipe_info['fat']\n",
+    "        recipe_info['nutrition']['proteins_100g'] = recipe_info['protein']\n",
+    "        recipe_info['nutrition']['carbohydrates_100g'] = recipe_info['carbohydrates']\n",
+    "        # The energy had been obtained from ChatGPT, but can be calculated with the following operation to give more precise information: kcal = 9 x fats + 4 x carbohydrates + 4 x proteins\n",
+    "        recipe_info['nutrition']['energy_100g'] = recipe_info['nutrition']['fat_100g']*9 + recipe_info['nutrition']['carbohydrates_100g']*4 + recipe_info['nutrition']['proteins_100g']*4\n",
+    "        print(recipe_name, recipe_info['nutrition'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We take into account what is generated by ChatGOT to calculate nutritional values\n",
+    "# We do this in each of the dishes and we do it in order to try to have more precise information.\n",
+    "# Energy is not taken into account since ChatGPT has given data that does not fit the aforementioned formula (kcal = 9 x fats + 4 x carbohydrates + 4 x proteins).\n",
+    "# For this reason, its value is recalculated.\n",
+    "for recipe_name, recipe_info in food_info.items():\n",
+    "    recipe_info['sugars'] = (recipe_info['nutrition']['sugars_100g'] + recipe_info['sugars'])/2\n",
+    "    recipe_info['fiber'] = (recipe_info['nutrition']['fiber_100g'] + recipe_info['fiber'])/2\n",
+    "    recipe_info['fat'] = (recipe_info['nutrition']['fat_100g'] + recipe_info['fat'])/2\n",
+    "    recipe_info['protein'] = (recipe_info['nutrition']['proteins_100g'] + recipe_info['protein'])/2\n",
+    "    recipe_info['carbohydrates'] = (recipe_info['nutrition']['carbohydrates_100g'] + recipe_info['carbohydrates'])/2\n",
+    "    recipe_info['kcal'] = recipe_info['fat']*9 + recipe_info['carbohydrates']*4 + recipe_info['protein']*4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We round nutritional values ​​so that there are no decimals\n",
+    "for recipe_name, recipe_info in food_info.items():\n",
+    "    recipe_info['sugars'] = round(recipe_info['sugars'])\n",
+    "    recipe_info['fiber'] = round(recipe_info['fiber'])\n",
+    "    recipe_info['fat'] = round(recipe_info['fat'])\n",
+    "    recipe_info['protein'] = round(recipe_info['protein'])\n",
+    "    recipe_info['carbohydrates'] = round(recipe_info['carbohydrates'])\n",
+    "    recipe_info['kcal'] = round(recipe_info['kcal'])"
+   ]
```

Comment:
```suggestion
    "# We round nutritional values ​​so that there are no decimals\n",
    "for recipe_name, recipe_info in food_info.items():\n",
    "    recipe_info['sugars'] = round((recipe_info['nutrition']['sugars_100g'] + recipe_info['sugars'])/2)\n",
    "    recipe_info['fiber'] = round((recipe_info['nutrition']['fiber_100g'] + recipe_info['fiber'])/2)\n",
    "    recipe_info['fat'] = round((recipe_info['nutrition']['fat_100g'] + recipe_info['fat'])/2)\n",
    "    recipe_info['protein'] = round((recipe_info['nutrition']['proteins_100g'] + recipe_info['protein'])/2)\n",
    "    recipe_info['carbohydrates'] = round((recipe_info['nutrition']['carbohydrates_100g'] + recipe_info['carbohydrates'])/2)\n",
    "    recipe_info['kcal'] = round(recipe_info['fat']*9 + recipe_info['carbohydrates']*4 + recipe_info['protein']*4)"
   ]
  },
```

You can round it directly, no need for a dedicated second block

Reply:
Done